src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -39,6 +41,18 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
         {
             AuthenticationResult authResult = null;
             ILoggerAdapter logger = AuthenticationRequestParameters.RequestContext.Logger;
+            
+            // Prime the scheme before any cache lookup if we already have a binding cert from a prior mint
+            if (AuthenticationRequestParameters.IsMtlsPopRequested &&
+                !(AuthenticationRequestParameters.AuthenticationScheme is MtlsPopAuthenticationOperation))
```

Comment:
Can you please explain to me why this 2nd check is needed? If `AuthenticationRequestParameters.IsMtlsPopRequested` is true, then why do we care if the authentication scheme is not `MtlsPopAuthenticationOperation`?